### ISSUE TITLE
_context.SaveChangesAsync() always return 0 when update

### ISSUE THREAD

**Issue by trungphungduc** [JonPSmith/EfCore.GenericEventRunner#6]
Hello sir,

My Context Setup as:
` public partial class Context : DbContextWithEvents<Context>`

Then i update data as:
`var res = await _context.SaveChangesAsync();`

=> Result: res always return 0.
-------
res return 1 when i setup Context as:
` public partial class Context : DbContext`

So is it intentionally or something else ?

<img width="751" alt="Screen Shot 2021-06-30 at 14 23 59" src="https://user-images.githubusercontent.com/42766976/123919606-4b413180-d9af-11eb-9804-15cb730c020d.png">

**Comment by JonPSmith** [comment]
That is strange, as I return the `int` result from `SaveChanges` and `SaveChangesAsync` . 

I have added a couple of [simple unit tests](https://github.com/JonPSmith/EfCore.GenericEventRunner/blob/master/Test/UnitTests/DataLayerTests/TestExampleDbContext.cs) and I get a result of 1. Can you confirm that the entity you added/updated is in the database?

**Comment by trungphungduc** [comment]
> That is strange, as I return the `int` result from `SaveChanges` and `SaveChangesAsync` .
> 
> I have added a couple of [simple unit tests](https://github.com/JonPSmith/EfCore.GenericEventRunner/blob/master/Test/UnitTests/DataLayerTests/TestExampleDbContext.cs) and I get a result of 1. Can you confirm that the entity you added/updated is in the database?

@JonPSmith 
Hello sir,
1. The data updated successfully in the database (and return `int result = 0`)
2. I am trying to do more test. So wait for me.

My Env:
- Dotnet Core 5.
- Use Postgresql

**Comment by trungphungduc** [comment]
@JonPSmith 
Hello sir,
I found the problems. Check this line:
```
var logs = services.RegisterGenericEventRunner(
                Assembly.GetAssembly(typeof(AddHistoryDangTinAfterEventHandle)));
```
Case 1: When i use this line: 
=> res = 0 when updated.

Case 2: When i comment this line:
=> res = 1 when updated.

If you need to test something else, feel free let me know.

<img width="730" alt="Screen Shot 2021-07-01 at 15 46 33" src="https://user-images.githubusercontent.com/42766976/124127185-a0f80580-daa5-11eb-9982-07eb7e7dff81.png">
<img width="751" alt="Screen Shot 2021-07-01 at 15 47 06" src="https://user-images.githubusercontent.com/42766976/124127192-a35a5f80-daa5-11eb-84b3-48a4dfdcfd73.png">
<img width="487" alt="Screen Shot 2021-07-01 at 15 47 29" src="https://user-images.githubusercontent.com/42766976/124127197-a48b8c80-daa5-11eb-8380-7cf0133cce62.png">

**Comment by JonPSmith** [comment]
Hi @trungphungduc 

I have checked this and calling `SaveChangesAsync` doesn't return the number of update, while a call to `SaveChanges` works properly. 

I will fix this bug, but not immediately as I have other things I want to do first. I will close this issue when I have issued a fixed version of this library.

**Comment by trungphungduc** [comment]
@JonPSmith 
Hello Sir,
When i use SaveChanges, i can not use EventHandles, Error:
`Could not find a AfterSave event handler for the event DangTinWithHistoryEvent. Their was a suitable async event handler available, but you didn't call SaveChangesAsync.`

I am in a dilemma now.

<img width="979" alt="Screen Shot 2021-07-06 at 15 07 02" src="https://user-images.githubusercontent.com/42766976/124565234-05cbab00-de6c-11eb-80ac-3aa2330a29e1.png">

**Comment by JonPSmith** [comment]
You need to read the [documentation on event handlers](https://github.com/JonPSmith/EfCore.GenericEventRunner/wiki/Creating-event-handlers#notes-on-event-handlers) especially the notes. If you read the error message it does give you a good hint that you need a sync event handler.

Please read the documentation and if you have any other questions open a new issue. This issue is about a bug on SvaeChangesAsync.

**Comment by JonPSmith** [comment]
Hi @trungphungduc,

I found the bug and fixed it - there is a new release 2.3.3 of GenericEventRunner.

Thanks for pointing that out.
